client/src/dolphin/dolphin_interface.rs

```diff
@@ -262,7 +269,7 @@ fn get_emulated_base_address(pid: u32) -> Option<usize> {
         Ok(maps) => maps,
     };
 
-    let map = maps.iter().find(|m| {
+    let map = maps.iter().rev().find(|m| {
```

Comment:
I think we ought to have a comment explaining this iterator. Something that mentions that we're interested in the most recent map with the appropriate name (at least I think that's what the order within the maps file represents)